netutils/bandwidth.py

```diff
@@ -0,0 +1,527 @@
+
+    Args:
+        speed (float): Speed in Tbps to convert to Kbps.
+
+    Returns:
+        float: Speed in Kbps converted from Tbps.
+    """
+    return speed * 1000000000
+
+
+def tbps_to_kbytes(speed: float) -> float:
+    """Method to convert `speed` from Tbps to KBps value.
+
+    Args:
+        speed (float): Speed in Tbps to convert to KBps.
+
+    Returns:
+        float: Speed in KBps converted from Tbps.
+    """
+    return speed * 125000000
+
+
+def tbps_to_mbps(speed: float) -> float:
+    """Method to convert `speed` from Tbps to Mbps value.
+
+    Args:
+        speed (float): Speed in Tbps to convert to Mbps.
+
+    Returns:
+        float: Speed in Mbps converted from Tbps.
+    """
+    return speed * 1000000
+
+
+def tbps_to_mbytes(speed: float) -> float:
+    """Method to convert `speed` from Tbps to MBps value.
+
+    Args:
+        speed (float): Speed in Tbps to convert to MBps.
+
+    Returns:
+        float: Speed in MBps converted from Tbps.
+    """
+    return speed * 125000
+
+
+def tbps_to_gbps(speed: float) -> float:
+    """Method to convert `speed` from Tbps to Gbps value.
+
+    Args:
+        speed (float): Speed in Tbps to convert to Gbps.
+
+    Returns:
+        float: Speed in Gbps converted from Tbps.
+    """
+    return speed * 1000
+
+
+def tbps_to_gbytes(speed: float) -> float:
+    """Method to convert `speed` from Tbps to GBps value.
+
+    Args:
+        speed (float): Speed in Tbps to convert to GBps.
+
+    Returns:
+        float: Speed in GBps converted from Tbps.
+    """
+    return speed * 125
+
+
+def name_to_kbits(speed: str) -> int:
+    """Method to convert a short bandwidth name to int value in Kbps.
+
+    Args:
+        speed (str): Bandwidth to be converted like `100Gbps` to Kbps.
+
+    Returns:
+        int: int value of bandwidth to be converted to Kbps
+    """
+    if re.search("[mM]bps", speed):
+        _value = int(mbps_to_kbps(int(re.sub("[mM]bps", "", speed))))
+    if re.search("[gG]bps", speed):
+        _value = int(gbps_to_kbps(int(re.sub("[gG]bps", "", speed))))
+    return _value
+
+
+def name_to_bits(speed: str) -> int:
+    """Method to convert a short bandwidth name to int value in bps.
+
+    Args:
+        speed (str): Bandwidth to be converted like `100Gbps` to bps.
+
+    Returns:
+        int: int value of bandwidth to be converted to bps
+    """
+    if re.search("[mM]bps", speed):
+        _value = name_to_kbits(speed) * 1000
+    if re.search("[gG]bps", speed):
+        _value = name_to_kbits(speed) * 1000
+    return _value
+
+
+def name_to_kbytes(speed: str) -> int:
+    """Method to convert a short bandwidth name to int value in KBps.
+
+    Args:
+        speed (str): Bandwidth to be converted like `100GBps` to KBps.
+
+    Returns:
+        int: int value of bandwidth to be converted to KBps
+    """
+    if re.search("[mM]Bps", speed):
+        _value = int(mbps_to_kbytes(int(re.sub("[mM]Bps", "", speed))))
```

Comment:
as far as. understand, this function will not be able to process `2.5Gbps` right now because we are only looking for an int

Since we are supporting decimals in `bits_to_name`, I think it would make sense to support decimal in all functions as well. it might be as simple as replacing all `int` with `float`

Reply:
I've updated to use float and adjust the tests. The biggest problem I see now is that all answers will have a decimal point even if it's a whole number. If we're OK with that I'll leave it as is.